include/nlohmann/json.hpp

```diff
@@ -5548,14 +5634,20 @@ class basic_json
             // check if iterator pos fits to this JSON value
             if (JSON_HEDLEY_UNLIKELY(pos.m_object != this))
             {
-                JSON_THROW(invalid_iterator::create(202, "iterator does not fit current value"));
+                JSON_THROW(invalid_iterator::create(202, "iterator does not fit current value", diagnostics_t(*this)));
             }
 
             // insert to array and return iterator
+#if JSON_DIAGNOSTICS
+            iterator result = insert_iterator(pos, val);
+            result->m_parent = this;
+            return result;
+#else
             return insert_iterator(pos, val);
+#endif
```

Comment:
So that there is no possibility of mismatch between #if/#else should we do the following?

    iterator result = insert_iterator(pos, val);
    set_parent(result);
    return result;

Reply:
This is the comment referred to as "As above" in the ones that actually appear above this one on the conversation view because I hadn't actually submitted this one when I submitted the other ones.

Reply:
Fixed with f8037660d06a0fbb6b1cde944d436032d04564ae.